leshan-server-core/src/main/java/org/eclipse/leshan/server/impl/SecurityRegistryImpl.java

```
@@ -185,6 +194,15 @@ protected void saveToFile() {
     }
 
     @Override
+    public void setTrustedCertificates(Certificate[] trustedCertificates) {
+        if (trustedCertificates != null && trustedCertificates.length > 0) {
+            this.trustedCertificates = trustedCertificates;
+        } else {
+            LOG.debug("Tried to set trusted certificates to null");
```

nasty silent error :) use Validate.notNull?